tests/native_type_test_impl.cpp

```diff
@@ -0,0 +1,25 @@
+#include "native_type_test_impl.h"
+
+// looks like VS10 does not support std::vector with explicit alignment
```

Comment:
No need for a .cpp, just inline in the .h

Reply:
It is required, otherwise we will have circular dependencies. 
native_type_test_generated.h requires `#include "native_type_test_impl.h"`
and native_type_test_impl.cpp requires `#include "native_type_test_generated.h"`